src/openfermion/transforms/_code_transform_functions.py

```diff
@@ -0,0 +1,254 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+""" Pre-existing codes for Fermion-qubit mappings
+    based on (arXiv:1712.07067) """
+
+import numpy
+
+from openfermion.ops import BinaryCode, SymbolicBinary, linearize_decoder
+
+
+def _encoder_bk(dimension):
```

Comment:
There is probably some obvious reason why you use "dimension" here and "modes" below?